### ISSUE TITLE
SSH host key generation failed

### ISSUE THREAD

**Issue by aismail14** [F5Networks/f5-azure-arm-templates#202]
Greeting,

I am facing the below error, however I did a successful deployment for the template with the same password previously.  

{"code":"DeploymentFailed","message":"At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.","details":[{"code":"Conflict","message":"{\r\n \"status\": \"Failed\",\r\n \"error\": {\r\n \"code\": \"ResourceDeploymentFailure\",\r\n \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n \"details\": [\r\n {\r\n \"code\": \"OSProvisioningClientError\",\r\n \"message\": \"SSH host key generation failed. Error details: [ProvisionError] Failed to generate ssh host key: ret=255, out= /config/ssh/ssh_host_rsa_key.pub is not a public key file.\\n. To resolve this issue verify if Linux agent is set up properly. \\r\\n You can check the instructions at : https://azure.microsoft.com/documentation/articles/virtual-machines-linux-agent-user-guide/\\r\\n \"\r\n }\r\n ]\r\n }\r\n}"}]}

## Do you already have an issue opened with F5 support?
Github Issues are consistently monitored by F5 staff, but should be considered as best effort only and you should not expect to receive the same level of response as provided by F5 Support. Please [open an case](https://support.f5.com/csp/article/K2633) with F5 if this is a critical issue.

## Description
Describe the problem you're having or the enhancement you'd like to request.

## Template
For bugs, enter the template with which you are experiencing issues below.

## Severity Level
For bugs, enter the bug severity level. Do not set any labels.

Severity: <Fill in level: 1 through 5>

Severity level definitions:
 1. **Severity 1 (Critical)** : Defect is causing systems to be offline and/or nonfunctional. immediate attention is required.
 2. **Severity 2 (High)** : Defect is causing major obstruction of system operations.
 3. **Severity 3 (Medium)** : Defect is causing intermittent errors in system operations.
 4. **Severity 4 (Low)** : Defect is causing infrequent interuptions in system operations.
 5. **Severity 5 (Trival)** : Defect is not causing any interuptions to system operations, but none-the-less is a bug.

**Comment by shyawnkarim** [comment]
@aismail14 are you still getting this error?

**Comment by jmcalalang** [comment]
looks like a duplicate of: https://github.com/F5Networks/f5-azure-arm-templates/issues/191

**Comment by aismail14** [comment]
@shyawnkarim yes on the latest version, I tried version 14 and it worked fine on the same instance type.

however I created a service request with the support and they refused to help as it is a new deployment !!! they just sent me PS numbers !!

@jmcalalang yes it is. I just tried an older version and it worked as I mentioned.

Thank you all.

**Comment by shyawnkarim** [comment]
The problem is that the Azure waagent tries to access the keys before they are generated. We are tracking this bug internally with VECLOUD-971.

**Comment by shyawnkarim** [comment]
Closing due to age. These legacy templates are now in maintenance mode and are being replaced by our next-generation templates available in the [Cloud Templates 2.0 GitHub repo](https://github.com/F5Networks/f5-azure-arm-templates-v2).
